api/v1alpha3/types.go

```diff
@@ -87,6 +87,9 @@ type VnetSpec struct {
 
 // IsManaged returns true if the vnet is managed.
 func (v *VnetSpec) IsManaged(clusterName string) bool {
+	if v.Tags == nil {
```

Comment:
I don't think this is needed as @nojnhuh pointed out

` Tags`  is a `map[string]string` , not a pointer to a map, so even if the map is nil it won't panic:

https://go.dev/play/p/-iu_DXwnIrE